requirements/extra-ml-requirements.txt

```diff
@@ -67,6 +67,7 @@ tiktoken
 tenacity
 # Required by mlflow.langchain
 langchain
+langchain_core
```

Comment:
BaseTransformOutputParser is from langchain_core https://api.python.langchain.com/en/latest/output_parsers/langchain_core.output_parsers.transform.BaseTransformOutputParser.html

Reply:
We don't need to add this, installing `langchain` should automatically install `lancghain-core`, too.
https://python.langchain.com/v0.1/docs/get_started/installation/#langchain-core

Reply:
Hmm. When I try this in practice it doesnt seem to work

```bash
(mlflow-dev-env) (base) prithvi.kannan@NR93P4PJF3 mlflow % pip uninstall langchain-core
Found existing installation: langchain-core 0.2.4
Uninstalling langchain-core-0.2.4:
  Would remove:
    /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages/langchain_core-0.2.4.dist-info/*
    /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages/langchain_core/*
Proceed (Y/n)? y
  Successfully uninstalled langchain-core-0.2.4
(mlflow-dev-env) (base) prithvi.kannan@NR93P4PJF3 mlflow % pip install langchain
pyRequirement already satisfied: langchain in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (0.0.296)
Requirement already satisfied: PyYAML>=5.3 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from langchain) (6.0)
Requirement already satisfied: SQLAlchemy<3,>=1.4 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from langchain) (2.0.8)
Requirement already satisfied: aiohttp<4.0.0,>=3.8.3 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from langchain) (3.8.4)
Requirement already satisfied: async-timeout<5.0.0,>=4.0.0 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from langchain) (4.0.2)
Requirement already satisfied: dataclasses-json<0.7,>=0.5.7 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from langchain) (0.6.0)
thCollecting langsmith<0.1.0,>=0.0.38 (from langchain)
  Using cached langsmith-0.0.92-py3-none-any.whl.metadata (9.9 kB)
Requirement already satisfied: numexpr<3.0.0,>=2.8.4 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from langchain) (2.8.6)
Requirement already satisfied: numpy<2,>=1 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from langchain) (1.23.3)
Requirement already satisfied: pydantic<3,>=1 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from langchain) (1.10.13)
Requirement already satisfied: requests<3,>=2 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from langchain) (2.28.2)
Requirement already satisfied: tenacity<9.0.0,>=8.1.0 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from langchain) (8.2.2)
Requirement already satisfied: attrs>=17.3.0 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from aiohttp<4.0.0,>=3.8.3->langchain) (22.2.0)
Requirement already satisfied: charset-normalizer<4.0,>=2.0 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from aiohttp<4.0.0,>=3.8.3->langchain) (3.1.0)
Requirement already satisfied: multidict<7.0,>=4.5 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from aiohttp<4.0.0,>=3.8.3->langchain) (6.0.4)
Requirement already satisfied: yarl<2.0,>=1.0 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from aiohttp<4.0.0,>=3.8.3->langchain) (1.8.2)
Requirement already satisfied: frozenlist>=1.1.1 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from aiohttp<4.0.0,>=3.8.3->langchain) (1.3.3)
Requirement already satisfied: aiosignal>=1.1.2 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from aiohttp<4.0.0,>=3.8.3->langchain) (1.3.1)
Requirement already satisfied: marshmallow<4.0.0,>=3.18.0 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from dataclasses-json<0.7,>=0.5.7->langchain) (3.20.1)
Requirement already satisfied: typing-inspect<1,>=0.4.0 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from dataclasses-json<0.7,>=0.5.7->langchain) (0.9.0)
oRequirement already satisfied: typing-extensions>=4.2.0 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from pydantic<3,>=1->langchain) (4.8.0)
Requirement already satisfied: idna<4,>=2.5 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from requests<3,>=2->langchain) (3.4)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from requests<3,>=2->langchain) (1.26.15)
Requirement already satisfied: certifi>=2017.4.17 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from requests<3,>=2->langchain) (2023.5.7)
Requirement already satisfied: packaging>=17.0 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from marshmallow<4.0.0,>=3.18.0->dataclasses-json<0.7,>=0.5.7->langchain) (23.2)
Requirement already satisfied: mypy-extensions>=0.3.0 in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (from typing-inspect<1,>=0.4.0->dataclasses-json<0.7,>=0.5.7->langchain) (1.0.0)
nDownloading langsmith-0.0.92-py3-none-any.whl (56 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 56.5/56.5 kB 1.3 MB/s eta 0:00:00
Installing collected packages: langsmith
  Attempting uninstall: langsmith
    Found existing installation: langsmith 0.1.71
    Uninstalling langsmith-0.1.71:
      Successfully uninstalled langsmith-0.1.71
Successfully installed langsmith-0.0.92

[notice] A new release of pip is available: 23.3 -> 24.0
[notice] To update, run: pip install --upgrade pip
(mlflow-dev-env) (base) prithvi.kannan@NR93P4PJF3 mlflow % python
Python 3.9.13 (main, Aug 25 2022, 18:24:45) 
[Clang 12.0.0 ] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import langchain_core
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'langchain_core'
```

Reply:
@prithvikannan I think it's because you already has `langchain` installed.
```
Requirement already satisfied: langchain in /Users/prithvi.kannan/opt/miniconda3/envs/mlflow-dev-env/lib/python3.9/site-packages (0.0.296)
```
Could you try again with uninstalling `langchain`, not only `langchain-core`

Reply:
ah ur right. that worked for me. removed `langchain-core` from requirements.